src/WebJobs.Script/Workers/IFunctionInvocationDispatcher.cs

```diff
@@ -19,6 +20,8 @@ public interface IFunctionInvocationDispatcher : IDisposable
 
         Task InitializeAsync(IEnumerable<FunctionMetadata> functions, CancellationToken cancellationToken = default);
 
+        Task<IDictionary<string, WorkerStatus>> GetWorkerStatusesAsync();
```

Comment:
Responsibility of this interface: IFunctionInvocationDispatcher.cs is to manage worker channels and distribute invocations.

I think  creating separate interface for GetWorkerStatusesAsync would be better. Also, implementing this interface is based on hosting environment will give us more control. 

- Can we skip this when running locally?
- Would the monitoring for processes differ when running in windows vs linux?

Reply:
The difficulty here is that while for Rpc, channels are registered with the external JobHostLanguageWorkerChannelManager, for HttpWorker there is no external registry. So for the former, we could have a new separate IRpcWorkerChannelStatusProvider that takes the channel managers on construction and implements GetWorkerStatusesAsyc by querying initialized channels. However, how would we do that for HttpWorkers? There isn't any channel manager - the channels are managed internally by HttpFunctionInvocationDispatcher.

So to keep things simple, I've exposed this functionality via the common service interface, which happens to be the dispatcher. Allowing the dispatcher to provide status info for the channels it manages seems OK to me.

Reply:
Good point on HttpWorker channels. I have to add support multiple channels for HttpWorker as well. If this indeed makes sense, I will look into as part of refactoring. Thanks!